### ISSUE TITLE
The Today Button is not accessible via Tab key for keyboard users

### ISSUE THREAD

**Issue by talatkuyuk** [Hacker0x01/react-datepicker#3885]
The problem is: The keyboard users can not focus to the Today Button (actually it is a `div`) via `Tab` key.

Today button's tab index (tabIndex) actually should be set to 0. Otherwise it is not accessible from keyboard.

https://github.com/Hacker0x01/react-datepicker/blob/master/src/calendar.jsx#L734

**Comment by marwanKettleCom** [comment]
Yes having the same issue.  Please fix.  Thank you!

**Comment by LGLabGreg** [comment]
There is no need to add a button as there is already logic to set tabindex, but currently it is possible that no date has tabindex=0 if the datepicker loads with no selected date.

https://github.com/Hacker0x01/react-datepicker/blob/master/src/day.jsx#L282

So if not keyboardSelected/selected/preselected it should default to today or first day of the month.

**Comment by LGLabGreg** [comment]
Here's a solution

```
const datePickerRef = useRef(null);

useEffect(() => {
  if (datePickerRef.current) {
    datePickerRef.current.setPreSelection(new Date());
  }
}, [datePickerRef]);

<DatePicker ref={datePickerRef} />
```

**Comment by talatkuyuk** [comment]
My solution is inline with `selectsRange` prop of the date picker. Update yours accordingly. 

I assume you use `todayButton="Today"` prop already to see Today button at the bottom of the date picker.

Firstly, use `onCalendarOpen={setupTodayButton}` as a prop.

```tsx
const setupTodayButton = () => {
    setTimeout(() => {
      const todayButton = document.querySelector(
        ".react-datepicker__today-button",
      ) as HTMLElement | null;
      if (!todayButton) return;

      todayButton.setAttribute("tabindex", "0");
      todayButton.setAttribute("role", "button");

      const applyTodayRange = () => {
        const startDate = new Date();
        const endDate = new Date();
        endDate.setHours(23, 59, 59, 999);
        handleChange([startDate, endDate]); // replace it with your state setter
      };

      todayButton.onclick = applyTodayRange;

      todayButton.onkeydown = (e) => {
        if (e.key === "Enter" || e.key === " ") {
          e.preventDefault();
          applyTodayRange();
        }
      };
    }, 0);
  };
```
If you use it as **inline** date picker, then you should use an useEffect, I suppose.
